ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/rules/HiveProjectSortExchangeTransposeRule.java

```diff
@@ -88,6 +88,11 @@ public void onMatch(RelOptRuleCall call) {
 
     RelCollation newCollation = RelCollationTraitDef.INSTANCE.canonize(RelCollationImpl.of(fieldCollations));
     List<Integer> newDistributionKeys = getNewRelDistributionKeys(project, sortExchange.getDistribution());
+    if (newDistributionKeys == null) {
```

Comment:
When there are no distribution keys the project can be safely pushed down because there are no columns/expressisons that must present in the project expression list.